fsspec/tests/test_spec.py

```diff
@@ -611,3 +1039,248 @@ def check_events(lpaths, rpaths):
     fs.get(base, dest, callback=callback)
     check_events(base, dest)
     callback.events.clear()
+
+
+def _clean_paths(paths, prefix=""):
+    """
+    Helper to cleanup paths results by doing the following:
+      - remove the prefix provided from all paths
+      - remove the trailing slashes from all paths
+      - remove duplicates paths
+      - remove empty paths after removing the prefix
+      - sort all paths
+    """
+    paths_list = paths
+    if isinstance(paths, dict):
+        paths_list = list(paths)
+    paths_list = [p.replace(prefix, "").strip("/") for p in paths_list]
+    paths_list = [p for p in sorted(set(paths_list)) if p]
+    if isinstance(paths, dict):
+        return {p: paths[p] for p in paths_list}
+    return paths_list
+
+
+@pytest.fixture(scope="function")
+def glob_fs():
+    return DummyTestFS(fs_content=PATHS_FOR_GLOB_TESTS)
+
+
+@pytest.mark.skipif(
+    sys.platform.startswith("win"), reason="can't run bash tests on windows"
+)
+@pytest.mark.parametrize(
+    ("path", "expected"),
+    GLOB_POSIX_TESTS,
+)
+def test_posix_tests(path, expected, tmp_path):
+    """
+    Tests against python glob and bash stat to check if our posix tests are accurate.
+    """
+    local_fs = LocalFileSystem(auto_mkdir=True)
+    local_fake_dir = str(tmp_path)
+    for path_info in PATHS_FOR_GLOB_TESTS:
+        if path_info["type"] == "file":
+            local_fs.touch(path=f"{str(tmp_path)}/{path_info['name']}")
+
+    abs_path = f"{local_fake_dir}/{path}"
+
+    # Test python glob
+    python_output = glob.glob(pathname=abs_path, recursive=True)
+    assert _clean_paths(python_output, local_fake_dir) == _clean_paths(expected)
+
+    # Test bash glob
+    bash_abs_path = (
+        abs_path.replace("\\", "\\\\")
+        .replace("$", "\\$")
+        .replace("(", "\\(")
+        .replace(")", "\\)")
+        .replace("|", "\\|")
+    )
+    bash_output = subprocess.run(
+        ["bash", "-c", f"shopt -s globstar && stat -c %N {bash_abs_path}"],
+        capture_output=True,
+    )
+    bash_output = bash_output.stdout.decode("utf-8").replace("'", "").split("\n")
+    assert _clean_paths(bash_output, local_fake_dir) == _clean_paths(expected)
```

Comment:
```suggestion
    try:
        bash_output = subprocess.run(
            ["bash", "-c", f"shopt -s globstar && stat -c %N {bash_abs_path}"],
            capture_output=True,
            check=True,
        )

        bash_output = bash_output.stdout.decode("utf-8").replace("'", "").split("\n")
        assert _clean_paths(bash_output, local_fake_dir) == _clean_paths(expected)
    except subprocess.CalledProcessError as e:
        # If bash or stat don't support the required options, skip this part
        # of the test.
        pass
```
We need to be careful here on macOS as the default versions of `bash` and `stat` that it ships with do not support `shopt` and the `-c` flag respectively. To avoid potential new contributors being put off by tests failing because of this, I think it best to skip this part of the test if the `subprocess.run` call fails.

Reply:
You're right, I split the posix tests between glob and bash tests and allow skipping bash tests if one of the following is true:
- the tests run on windows
- bash is not found
- globstar option is not found

BTW, those tests were useful for me while developing, to understand how the double asterisk works in bash and python glob. But we could also remove it completely since both of them don't actually test the `fsspec` library behavior.